app/controllers/api/v1/users/registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class RegistrationsController < Devise::RegistrationsController
+        respond_to :json
+
+        before_action :ensure_valid_client
+
+        def create
+          user = User.from_email(user_params[:email], user_params[:password])
+          user.errors.empty? ? render_success(user) : render_error(user)
+        end
+
+        private
+
+        def user_params
+          params.require(:user).permit(:email, :password)
+        end
+
+        def render_success(user)
+          data = UserSerializer.new(user).serializable_hash[:data]
+          message = I18n.t('activemodel.notices.models.user.create')
+
+          render json: { data: data, meta: { message: message } }, status: :created
+        end
+
+        def render_error(user)
+          render json: { errors: user.errors.full_messages }, status: :unprocessable_entity
```

Comment:
The current error responses are incorrect 🙈 . The format should follow the [JSON:API specs](https://jsonapi.org/format/#document-top-level):

> errors: an array of [error objects](https://jsonapi.org/format/#errors).

Reply:
Hmmm, which one of these fields do you recommend us to return? 💭 cc @hoangnguyen92dn 

![Screen Shot 2023-07-11 at 14 21 06](https://github.com/nimblehq/ic-rails-huey-toby/assets/18277915/72b55d7c-c8c4-43a2-8d2a-b5c500a39c40)

Reply:
@hoangmirs Oh, I think @hoangnguyen92dn has provided me the answer [here](https://github.com/nimblehq/ic-rails-huey-toby/pull/44#discussion_r1259156268) 👍

Reply:
@hoangnguyen92dn Should I add a TODO comment and leave it for you to update it? 🙏

Reply:
@Tuubz That's fine. We could update in the next upcoming tasks 👍